File: server/use-cases/getServiceOrderCount.ts
import prisma from "../utils/prisma"
import { weekYearToString } from "../utils/weekTools"

import {
    GetServiceOrderCountParamsType,
} from '../../schemas/preventive'

export async function getServiceOrderCount({ week, year }: GetServiceOrderCountParamsType) {
    try {
        const weekCode = weekYearToString(week, year)

        const finished = await prisma.preventiveOS.count({
            where: {
                weekCode,
                concluded: true
            }
        })
        const unfinished = await prisma.preventiveOS.count({
            where: {
                weekCode,
                concluded: false
            }
        })

        return {
            finished, unfinished
        }
    } catch (error) {
        throw error
    }
}

File: src-py/use_cases/get_service_order_count.py
from schemas.preventive import GetServiceOrderCountParams, ServiceOrderCount
from infra.prisma import prisma
from utils.week_tools import weekYearToString


async def getServiceOrderCount(params: dict) -> ServiceOrderCount:
    validated = GetServiceOrderCountParams.model_validate(params)
    week_code = weekYearToString(validated.week, validated.year)

    finished = await prisma.preventiveos.count(
        where={"weekCode": week_code, "concluded": True},
    )
    unfinished = await prisma.preventiveos.count(
        where={"weekCode": week_code, "concluded": False},
    )

    return ServiceOrderCount(finished=finished, unfinished=unfinished)
